### PR TITLE
Integration test refactors to use dockerCmd

### PR DESCRIPTION
**Pull request by vdemeester**
Part of #14603, this refactor the following integration-cli test files (and api too) to use as much as possible `dockerCmd`.
- integration-cli/docker_cli_attach_test.go
- integration-cli/docker_cli_attach_unix_test.go
- integration-cli/docker_cli_build_test.go
- integration-cli/docker_cli_build_unix_test.go
- integration-cli/docker_cli_by_digest_test.go
- integration-cli/docker_cli_commit_test.go
- integration-cli/docker_cli_config_test.go
- integration-cli/docker_cli_cp_test.go
- integration-cli/docker_cli_create_test.go
- integration-cli/docker_cli_pause_test.go
- integration-cli/docker_cli_port_test.go
- integration-cli/docker_cli_port_unix_test.go
- integration-cli/docker_cli_proxy_test.go
- integration-cli/docker_cli_ps_test.go
- integration-cli/docker_cli_pull_test.go
- integration-cli/docker_cli_push_test.go
- integration-cli/docker_api_attach_test.go
- integration-cli/docker_api_containers_test.go
- integration-cli/docker_api_events_test.go
- integration-cli/docker_api_exec_resize_test.go
- integration-cli/docker_api_exec_test.go
- integration-cli/docker_api_images_test.go
- integration-cli/docker_api_info_test.go

I think I got it all on my part of the issue :sweat_smile:. Depending on when #14623 will be ready and merged, some rebasing on my part might be needed.

### CODE REVIEW

**Comment by HuKeping** [comment]
feel free to go first and I can rebase mine, `integration-cli/docker_cli_run_test.go` is a huge file and it will cost some time :(

File: integration-cli/docker_cli_build_test.go [hunk 1]
```diff
@@ -420,12 +411,12 @@ func (s *DockerSuite) TestBuildEnvEscapes(c *check.C) {
     `,
 		true)
 
-	out, _, err := runCommandWithOutput(exec.Command(dockerBinary, "run", "-t", name))
-
 	if err != nil {
```
Comment (cpuguy83) [inline integration-cli/docker_cli_build_test.go]: Did we miss this?

File: integration-cli/docker_cli_create_test.go [hunk 2]
```diff
@@ -297,47 +251,25 @@ func (s *DockerSuite) TestCreateRM(c *check.C) {
...
-	if err != nil {
-		c.Fatal(err, out)
-	}
+	out, _ := dockerCmd(c, "create", "busybox")
 	id := strings.TrimSpace(out)
 
-	cmd = exec.Command(dockerBinary, "create", fmt.Sprintf("--ipc=container:%s", id), "busybox")
-	out, _, err = runCommandWithOutput(cmd)
-	if err != nil {
+	if out, _, err := dockerCmdWithError(c, "create", fmt.Sprintf("--ipc=container:%s", id), "busybox"); err != nil {
```
Comment (cpuguy83) [inline integration-cli/docker_cli_create_test.go]: Doesn't seem like this one needs `WithError`

File: integration-cli/docker_cli_by_digest_test.go [hunk 3]
```diff
@@ -40,16 +35,14 @@ func setupImageWithTag(tag string) (string, error) {
...
+	out, _, err := dockerCmdWithError(c, "push", repoAndTag)
 	if err != nil {
 		return "", fmt.Errorf("pushing the image to the private registry has failed: %s, %v", out, err)
 	}
 
 	// delete our local repo that we previously tagged
-	cmd = exec.Command(dockerBinary, "rmi", repoAndTag)
-	if out, _, err := runCommandWithOutput(cmd); err != nil {
-		return "", fmt.Errorf("error deleting images prior to real test: %s, %v", out, err)
+	if rmiout, _, err := dockerCmdWithError(c, "rmi", repoAndTag); err != nil {
```
Comment (LK4D4) [inline integration-cli/docker_cli_by_digest_test.go]: What the point of this function? :) looks like it only complicates code.

File: integration-cli/docker_cli_build_test.go [hunk 4]
```diff
@@ -3371,8 +3320,8 @@ func (s *DockerSuite) TestBuildVerifyIntString(c *check.C) {
   MAINTAINER 123
   `, true)
 
-	out, rc, err := runCommandWithOutput(exec.Command(dockerBinary, "inspect", name))
-	if rc != 0 || err != nil {
+	out, rc := dockerCmd(c, "inspect", name)
```
Comment (cpuguy83) [inline integration-cli/docker_cli_build_test.go]: Isn't `rc` always 0 here?

**Comment by cpuguy83** [comment]
For cases where we need to supply some extra comment on an error for `dockerCmd`, this can be extended to take either `comments ..string`, and we can convert that to a `check.Commentf`.

File: integration-cli/docker_cli_build_test.go (hunk 1 above)
Comment (vdemeester) [inline integration-cli/docker_cli_build_test.go]: Oh I forgot to remove the if err != nil … indeed :sweat:

File: integration-cli/docker_cli_by_digest_test.go (hunk 3 above)
Comment (vdemeester) [inline integration-cli/docker_cli_by_digest_test.go]: ``dockerCmdWithError`` or ``setupImageWithTag`` ? I'm note sure in what we it complicates the code that much though :blush:.

**Comment by vdemeester** [comment]
@cpuguy83 I'll take a look at that :wink:.

**Comment by calavera** [comment]
@vdemeester @cpuguy83 I'd rather not add any logic that we don't need right now. It will stop being a refactor if we do so.

**Comment by cpuguy83** [comment]
@calavera The point was to not have to use `dockerCmdWithError` just because we want a custom error message.

File: integration-cli/docker_cli_build_test.go (hunk 1 above)
Comment (vdemeester) [inline integration-cli/docker_cli_build_test.go]: Oh actually, no check was done on the ``err`` out of the ``buildImage`` just a little *up*, that's probably why I didn't remove the check on err.

File: integration-cli/docker_cli_build_test.go (hunk 4 above)
Comment (vdemeester) [inline integration-cli/docker_cli_build_test.go]: That's right >.<'

**Comment by calavera** [comment]
oh, I see. I kinda like better to be explicit about calling another function when you want to check the error.

**Comment by calavera** [comment]
LGTM

**Comment by cpuguy83** [comment]
LGTM

**Pull request merged**